drivers/ethernet/eth_mcux.c

```diff
@@ -286,12 +287,14 @@ static void eth_mcux_phy_event(struct eth_context *context)
 					  kENET_MiiReadValidFrame);
 			context->link_up = link_up;
 			context->phy_state = eth_mcux_phy_state_read_duplex;
+			ethernet_mgmt_raise_carrier_on_event(context->iface);
```

Comment:
Hum, I am not super found of raising net_mgmt events from device drivers directly. (only in very particular case as in offloaded drivers, and still: it does not do it directly).

Instead, create 2 ethernet public functions implemented in ethernet l2 side:  ethernet_carrier_on()  and ethernet_carrier_off() 
There you could handle what you already do in the l2 side (calling net_if_up(), etc... and raise the events).
At least, form drivers/l2/net_if point of view it's reliable that way. (let's avoid using net_mgmt for core features, unless we are really forced to.)

That said, it helped to find out an issue in net_mgmt: it was not possible to raise an event in ISR mode.